pkg/e2etemplates/aws_operator_chart_values_test.go

```diff
@@ -0,0 +1,297 @@
+          Enabled: true
+        Encrypter: 'vault'
+    Registry:
+      Domain: quay.io
+    Secret:
+      AWSOperator:
+        IDRSAPub: test-idrsa-pub
+        SecretYaml: |
+          service:
+            aws:
+              accesskey:
+                id: test-access-key-id
+                secret: test-access-key-secret
+                token: test-access-key-token
+              hostaccesskey:
+                id: test-host-access-key-id
+                secret: test-host-access-key-secret
+                token: test-host-access-key-token
+
+      Registry:
+        PullSecret:
+          DockerConfigJSON: "{\"auths\":{\"quay.io\":{\"auth\":\"registry-pull-secret\"}}}"
+    Security:
+      RestrictAccess:
+        Enabled: false
+`,
+			errorMatcher: nil,
+		},
+		{
+			name: "case 2: all optional values left",
+			data: newAWSOperatorChartAllSetFromFullySet(func(v *AWSOperatorChart) {
+				v.Guest.Update.Enabled = false
+				v.Provider.AWS.Encrypter = ""
+			}),
+			expectedValues: `Installation:
+  V1:
+    Auth:
+      Vault:
+        Address: http://vault.default.svc.cluster.local:8200
+    Guest:
+      Kubernetes:
+        API:
+          Auth:
+            Provider:
+              OIDC:
+                ClientID: ""
+                IssueURL: ""
+                UsernameClaim: ""
+                GroupsClaim: ""
+      SSH:
+        SSOPublicKey: 'test'
+      Update:
+        Enabled: false
+    Name: ci-aws-operator
+    Provider:
+      AWS:
+        Region: 'eu-central-1'
+        DeleteLoggingBucket: true
+        IncludeTags: true
+        Route53:
+          Enabled: true
+        Encrypter: 'kms'
+    Registry:
+      Domain: quay.io
+    Secret:
+      AWSOperator:
+        IDRSAPub: test-idrsa-pub
+        SecretYaml: |
+          service:
+            aws:
+              accesskey:
+                id: test-access-key-id
+                secret: test-access-key-secret
+                token: test-access-key-token
+              hostaccesskey:
+                id: test-host-access-key-id
+                secret: test-host-access-key-secret
+                token: test-host-access-key-token
+
+      Registry:
+        PullSecret:
+          DockerConfigJSON: "{\"auths\":{\"quay.io\":{\"auth\":\"registry-pull-secret\"}}}"
+    Security:
+      RestrictAccess:
+        Enabled: false
+`,
+			errorMatcher: nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			values, err := Render(AWSOperatorChartValues, tc.data)
+
+			switch {
+			case err == nil && tc.errorMatcher == nil:
+				// correct; carry on
+			case err != nil && tc.errorMatcher == nil:
+				t.Fatalf("error == %#v, want nil", err)
+			case err == nil && tc.errorMatcher != nil:
+				t.Fatalf("error == nil, want non-nil")
+			case !tc.errorMatcher(err):
+				t.Fatalf("error == %#v, want matching", err)
+			}
+
+			if tc.errorMatcher != nil {
+				return
+			}
+
+			line, difference := diff(values, tc.expectedValues)
```

Comment:
`diff` is just test scaffolding function. It's written in helper_test.go.
W/o it we would have to narrow down long rendered outputs. This one gives the line number and prints two differing lines.